sympy/assumptions/ask.py

```diff
@@ -9,66 +9,1158 @@
+        return Predicate('positive')
+
+    @predicate_memo
+    def negative(self):
+        r"""
+        Negative number predicate.
+
+        ``Q.negative(x)`` is true iff ``x`` is a real number and :math:`x < 0`, that is,
+        it is in the interval :math:`(-\infty, 0)`.  Note in particular that negative
+        infinity is not negative.
+
+        A few important facts about negative numbers:
+
+        - Note that ``Q.nonnegative`` and ``~Q.negative`` are *not* the same
+          thing. ``~Q.negative(x)`` simply means that ``x`` is not negative,
+          whereas ``Q.nonnegative(x)`` means that ``x`` is real and not
+          negative, i.e., ``Q.nonnegative(x)`` is logically equivalent to
+          ``Q.zero(x) | Q.positive(x)``.  So for example, ``~Q.negative(I)`` is
+          true, whereas ``Q.nonnegative(I)`` is false.
+
+        - See the documentation of ``Q.real`` for more information about
+          related facts.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, symbols, I
+        >>> x = symbols('x')
+        >>> ask(Q.negative(x), Q.real(x) & ~Q.positive(x) & ~Q.zero(x))
+        True
+        >>> ask(Q.negative(-1))
+        True
+        >>> ask(Q.nonnegative(I))
+        False
+        >>> ask(~Q.negative(I))
+        True
+
+        """
+        return Predicate('negative')
+
+    @predicate_memo
+    def zero(self):
+        """
+        Zero number predicate.
+
+        ``ask(Q.zero(x))`` is true iff the value of ``x`` is zero.
+
+        Examples
+        ========
+
+        >>> from sympy import ask, Q, oo, symbols
+        >>> x, y = symbols('x, y')
+        >>> ask(Q.zero(0))
+        True
+        >>> ask(Q.zero(1/oo))
+        True
+        >>> ask(Q.zero(0*oo))
+        False
+        >>> ask(Q.zero(1))
+        False
+        >>> ask(Q.zero(x*y), Q.zero(x) | Q.zero(y))
+        True
+
+        """
+        return Predicate('zero')
+
+    @predicate_memo
+    def nonzero(self):
+        """
+        Nonzero real number predicate.
+
+        ``ask(Q.nonzero(x))`` is true iff ``x`` is real and ``x`` is not zero.  Note in
+        particular that ``Q.nonzero(x)`` is false if ``x`` is not real.  Use
+        ``~Q.zero(x)`` if you want the negation of being zero without any real
+        assumptions.
+
+        A few important facts about nonzero numbers:
+
+        - ``Q.nonzero`` is logically equivalent to ``Q.positive | Q.negative``.
+
+        - See the documentation of ``Q.real`` for more information about
+          related facts.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, symbols, I
+        >>> x = symbols('x')
+        >>> print(ask(Q.nonzero(x), ~Q.zero(x)))
+        None
+        >>> ask(Q.nonzero(x), Q.positive(x))
+        True
+        >>> ask(Q.nonzero(x), Q.zero(x))
+        False
+        >>> ask(Q.nonzero(0))
+        False
+        >>> ask(Q.nonzero(I)) # doctest: +SKIP
+        False
+        >>> ask(~Q.zero(I))
+        True
+
```

Comment:
I think we might want an example with infinity. So, one can see that it doesn't count as nonzero.